routers/user/auth.go

```diff
@@ -120,6 +127,11 @@ func SignInPost(ctx *context.Context, form auth.SignInForm) {
 	// Clear whatever CSRF has right now, force to generate a new one
 	ctx.SetCookie(setting.CSRFCookieName, "", -1, setting.AppSubUrl)
 
+	if redirectTo := ctx.Query("redirect_to"); len(redirectTo) > 0 {
```

Comment:
I don't understand, the `sign in` button already contains `redirect_to` link and set to cookie in `func SignIn(ctx *context.Context)`